src/m_menu.c

```diff
@@ -4943,9 +4943,18 @@ static boolean M_ShortcutResponder(void)
 
     if (M_InputActivated(input_mouselook))
     {
-        mouselook = !mouselook;
-        padlook = !padlook;
-        togglemsg("Free Look %s", mouselook ? "On" : "Off");
+        if (ev->type == ev_joyb_down)
+        {
+            // Gamepad free look toggle only affects gamepad.
+            padlook = !padlook;
+            togglemsg("Gamepad Free Look %s", padlook ? "On" : "Off");
+        }
+        else
+        {
+            // Keyboard or mouse free look toggle only affects mouse.
+            mouselook = !mouselook;
+            togglemsg("Free Look %s", mouselook ? "On" : "Off");
+        }
```

Comment:
This means a keyboard binding for free look only affects the mouse. Seems like a safe assumption.

Reply:
Indeed, seems legit.